File: clients/ruby/grpc.rb
require './table_pb'
require 'grpc'
require './table_services_pb'

if ARGV.length != 4
    puts 'wrong number of arguments supplied, should be 4, serverIP, tableID1, tableID2, dbID'
    return
end


host = ARGV[0]
id1 = ARGV[1]
id2 = ARGV[2]
dbId = ARGV[3]

stub = TableProductProvider::Stub.new(host + ':9090', :this_channel_is_insecure)

# stub.GetTableProduct(Filter.new('6162cf849d5b528868c1b14d', '6162c07fd22ed745bb63611c', '6162bfa4d22ed745bb63611b'))
res = stub.get_table_product(Filter.new(id1: id1,id2: id2, dbId:dbId))

puts res

File: clients/grpc.go
package clients

import (
	"context"
	"fmt"
	table "github.com/nikitasadok/database-system/table/delivery/grpc"
	"google.golang.org/grpc"
	"os"
	"time"
)

func TablesProduct(host string, port int) {
	args := os.Args[2:]

	if len(args) != 3 {
		fmt.Println("wrong number of arguments supplied, should be 3. tableID1, tableID2, dbID")
		return
	}

	id1 := args[0]
	id2 := args[1]
	dbID := args[2]

	var opts []grpc.DialOption
	opts = append(opts, grpc.WithInsecure())
	conn, err := grpc.Dial(host + ":"+  fmt.Sprint(port), opts...)
	if err != nil {
		fmt.Println(err)
	}
	defer conn.Close()

	client := table.NewTableProductProviderClient(conn)
	res, err := client.GetTableProduct(context.Background(), &table.Filter{
		Id1:  id1,
		Id2:  id2,
		DbId: dbID,
	})
	time.Sleep(1 * time.Second)
	// fmt.Printf("%+v\n", res)
	for i, r := range res.Rows {
		fmt.Println("ROW", i)
		for _, c := range r.Cells {
			fmt.Printf("%+v\n", c)
		}
	}
}
